src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/umbblockgridsortable.directive.js

```diff
@@ -0,0 +1,705 @@
+
+                    const foundElIndex = orderedContainerElements.indexOf(foundEl);
+                    const placeAt = (placeAfter ? foundElIndex+1 : foundElIndex);
+
+                    move(orderedContainerElements, placeAt);
+
+                    return;
+                }
+                // We skipped the above part cause we are above or below container:
+
+                // Indication if drop is good:
+                if(updateAllowIndication(currentContainerVM, currentItem) === false) {
+                    return;
+                }
+                
+                if(dragY < currentContainerRect.top) {
+                    move(orderedContainerElements, 0);
+                } else if(dragY > currentContainerRect.bottom) {
+                    move(orderedContainerElements, -1);
+                }
+            }
+
+            function move(orderedContainerElements, newElIndex) {
+
+                newElIndex = newElIndex === -1 ? orderedContainerElements.length : newElIndex;
+                
+                const placeBeforeElement = orderedContainerElements[newElIndex];
+                if (placeBeforeElement) {
+                    // We do not need to move this, if the element to be placed before is it self.
+                    if(placeBeforeElement !== currentElement) {
+                        currentContainerElement.insertBefore(currentElement, placeBeforeElement);
+                    }
+                } else {
+                    currentContainerElement.appendChild(currentElement);
+                }
+
+                if(config.onChange) {
+                    config.onChange();
+                }
+            }
+
+
+            /** Removes an element from container and returns its items-data entry */
+            vm.getItemOfElement = function (element) {
+                if(!element) {
+                    return null;
+                }
+                return vm.items.find(entry => config.compareElementToModel(element, entry));
+            }
+            vm.removeItem = function (item) {
+                if(!item) {
+                    return null;
+                }
+                const oldIndex = vm.items.indexOf(item);
+                if(oldIndex !== -1) {
+                    return vm.items.splice(oldIndex, 1)[0];
+                }
+                return null;
+            }
+
+            vm.sync = function(element, fromVm) {
+
+                const movingItem = fromVm.getItemOfElement(element);
+                if(!movingItem) {
+                    console.error("Could not find item of sync item")
+                    return false;
+                }
+                if(vm.notifyRequestDrop({item: movingItem}) === false) {
+                    console.error("was not allowed for drop on this location.")
+                    return false;
+                }
+                if(fromVm.removeItem(movingItem) === null) {
+                    console.error("Sync could not remove item")
```

Comment:
Consider if we should warn the user for any of these errors

Reply:
Actually, I will remove these. They should not be considered as errors. This was more for debugging so can go away now.